File: pe-core/src/main/java/org/saar/core/renderer/RenderingOutput.java
package org.saar.core.renderer;

import org.saar.core.screen.MainScreen;
import org.saar.core.screen.Screen;
import org.saar.lwjgl.opengl.textures.ReadOnlyTexture;

public interface RenderingOutput {

    void to(Screen screen);

    ReadOnlyTexture toTexture();

    default void toMainScreen() {
        to(MainScreen.INSTANCE);
    }

}

File: pe-core/src/main/java/org/saar/core/renderer/RenderingOutput.kt
package org.saar.core.renderer

import org.saar.core.renderer.renderpass.RenderPassBuffers
import org.saar.core.screen.MainScreen
import org.saar.core.screen.Screen

class RenderingOutput<T : RenderPassBuffers>(
    private val screen: Screen,
    val buffers: T) {

    fun to(screen: Screen) = this.screen.copyTo(screen)

    fun toMainScreen() = to(MainScreen)
}